File: 61-rotate-list/rotate-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* rotateRight(struct ListNode* head, int k) {
    if(!head||!head->next||k==0) return head;
    int len=1;
   struct ListNode*temp=head;
    while(temp->next){
        temp=temp->next;
          len++;
    }
    temp->next=head;
    k=k%len;
    int steps=len-k;
    struct ListNode*ntail=head;
    for(int i=1;i<steps;i++){
        ntail=ntail->next;
    }
    struct ListNode*newhead=ntail->next;
    ntail->next=NULL;
    return newhead;
}

File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def rotateRight(self, head, k):
        if not head or not head.next or k == 0:
            return head 
        tail=head
        length=1
        while tail.next:
            tail=tail.next
            length+=1
        tail.next=head    
        k=k%length
        stepstotail=length-k-1
        newtail=head
        for _ in range(stepstotail):
            newtail=newtail.next 
        newhead=newtail.next
        newtail.next=None
        return newhead

          

